diff --git a/vboxsf/Makefile.include.header b/vboxsf/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxsf/Makefile.include.header
+++ b/vboxsf/Makefile.include.header
@@ -136,9 +136,6 @@
 endif # neq($(KERNELRELEASE),)
 
 # Kernel build folder
-ifeq ($(KERN_DIR),)
- KERN_DIR := /lib/modules/$(KERN_VER)/build
-endif
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> (currently $(KERN_VER)) and run Make again)

diff --git a/vboxguest/Makefile.include.header b/vboxguest/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxguest/Makefile.include.header
+++ b/vboxguest/Makefile.include.header
@@ -136,9 +136,6 @@
 endif # neq($(KERNELRELEASE),)
 
 # Kernel build folder
-ifeq ($(KERN_DIR),)
- KERN_DIR := /lib/modules/$(KERN_VER)/build
-endif
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> (currently $(KERN_VER)) and run Make again)

diff --git a/vboxvideo/Makefile.include.header b/vboxvideo/Makefile.include.header
index 8df1eb4d25..5a3e5604e7 100644
--- a/vboxvideo/Makefile.include.header
+++ b/vboxvideo/Makefile.include.header
@@ -136,9 +136,6 @@
 endif # neq($(KERNELRELEASE),)
 
 # Kernel build folder
-ifeq ($(KERN_DIR),)
- KERN_DIR := /lib/modules/$(KERN_VER)/build
-endif
 ifneq ($(shell if test -d $(KERN_DIR); then echo yes; fi),yes)
  $(error Error: unable to find the headers of the Linux kernel to build against. \
           Specify KERN_VER=<version> (currently $(KERN_VER)) and run Make again)
